topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -570,100 +547,63 @@ This will set everything up to use the function. We have:
->    --- /dev/null
->    +++ b/files/galaxy/dynamic_job_rules/map_resources.py
->    @@ -0,0 +1,42 @@
->    +import logging
->    +from galaxy.jobs.mapper import JobMappingException
->    +
->    +log = logging.getLogger(__name__)
->    +
->    +DESTINATION_IDS = {
->    +    1 : 'slurm',
->    +    2 : 'slurm-2c'
->    +}
->    +FAILURE_MESSAGE = 'This tool could not be run because of a misconfiguration in the Galaxy job running system, please report this error'
->    +
->    +
->    +def dynamic_cores_time(app, tool, job, user_email):
->    +    destination = None
->    +    destination_id = 'slurm'
->    +
->    +    # build the param dictionary
->    +    param_dict = job.get_param_values(app)
->    +
->    +    if param_dict.get('__job_resource', {}).get('__job_resource__select') != 'yes':
->    +        log.info("Job resource parameters not seleted, returning default destination")
->    +        return destination_id
->    +
->    +    # handle job resource parameters
->    +    try:
->    +        # validate params
->    +        cores = int(param_dict['__job_resource']['cores'])
->    +        time = int(param_dict['__job_resource']['time'])
->    +        destination_id = DESTINATION_IDS[cores]
->    +        destination = app.job_config.get_destination(destination_id)
->    +        # set walltime
->    +        if 'nativeSpecification' not in destination.params:
->    +            destination.params['nativeSpecification'] = ''
->    +        destination.params['nativeSpecification'] += ' --time=%s:00:00' % time
->    +    except:
->    +        # resource param selector not sent with tool form, job_conf.yml misconfigured
->    +        log.warning('(%s) error, keys were: %s', job.id, param_dict.keys())
->    +        raise JobMappingException(FAILURE_MESSAGE)
->    +
->    +    log.info('returning destination: %s', destination_id)
->    +    log.info('native specification: %s', destination.params.get('nativeSpecification'))
->    +    return destination or destination_id
+>    --- a/files/galaxy/config/tpv_rules_local.yml
+>    +++ b/files/galaxy/config/tpv_rules_local.yml
+>    @@ -6,6 +6,8 @@ tools:
+>         abstract: true
+>         cores: 1
+>         mem: cores * 4
+>    +    params:
+>    +      walltime: 8
+>       testing:
+>         cores: 2
+>         rules:
+>    @@ -17,6 +19,13 @@ tools:
+>               # last line in block must evaluate to a value - which determines whether the TPV if conditional matches or not
+>               user_email not in admin_users
+>             fail: Unauthorized. Only admins can execute this tool.
+>    +      - id: resource_params_defined
+>    +        if: |
+>    +          param_dict = job.get_param_values(app)
+>    +          param_dict.get('__job_resource', {}).get('__job_resource__select') == 'yes'
+>    +        cores: int(job.get_param_values(app)['__job_resource']['cores'])
+>    +        params:
+>    +           walltime: int(job.get_param_values(app)['__job_resource']['time'])
```

Comment:
```suggestion
>    +           walltime: "{int(job.get_param_values(app)['__job_resource']['time'])}"
```

Reply:
before wrapping this in "{}" the value of walltime is saved as the string: "int(job.get_param_values(app)['__job_resource']['time'])"